kotsadm/web/src/components/apps/AppSnapshotDetail.jsx

```diff
@@ -142,15 +143,17 @@ class AppSnapshotDetail extends Component {
 
       const snapshotDetails = response.backupDetail;
 
+      let series = [];
       if (!isEmpty(snapshotDetails?.volumes)) {
         if (snapshotDetails?.hooks && !isEmpty(snapshotDetails?.hooks)) {
-          this.setState({ series: this.getSeriesData([...snapshotDetails?.volumes, ...snapshotDetails?.hooks].sort((a, b) => new Date(a.started) - new Date(b.started))) })
+          series = this.getSeriesData([...snapshotDetails?.volumes, ...snapshotDetails?.hooks].sort((a, b) => new Date(a.started) - new Date(b.started)));
         } else {
-          this.setState({ series: this.getSeriesData((snapshotDetails?.volumes).sort((a, b) => new Date(a.started) - new Date(b.started))) })
+          series = this.getSeriesData((snapshotDetails?.volumes).sort((a, b) => new Date(a.started) - new Date(b.started)));
         }
       } else if ((snapshotDetails?.hooks && !isEmpty(snapshotDetails?.hooks))) {
-        this.setState({ series: this.getSeriesData((snapshotDetails?.hooks).sort((a, b) => new Date(a.started) - new Date(b.started))) })
+        series = this.getSeriesData((snapshotDetails?.hooks).sort((a, b) => new Date(a.started) - new Date(b.started)));
       }
+      this.setState({series: series});
```

Comment:
include this in the next `setState` statement on line 158 so that the UI does not render twice

Reply:
that would be cleaner although isnt the point of react state that it queues up state changes and renders on tick?

Reply:
i've never heard of that, but i'm pretty sure the best practice is to not have consecutive `setState` statements

Reply:
ill fix it but fyi https://stackoverflow.com/a/33613918/9599057